tests/test_make_whitelist.py

```diff
@@ -5,14 +5,13 @@
 
 
 @pytest.fixture
-def check_whitelist(v, capsys):
+def check_whitelist(v):
     def examine(code, results_before, results_after):
         v.scan(code)
         check(v.get_unused_code(), results_before)
-        capsys.readouterr()  # Clear captured text
-        v.make_whitelist()
-        whitelist = capsys.readouterr().out
-        v.scan(whitelist)
+        whitelist = [item.get_whitelist_string() for item in v.get_unused_code()]
+        for whitelist_string in whitelist:
+            v.scan(whitelist_string)
```

Comment:
One for-loop should be enough, no need for the generator list.

Reply:
What was I even thinking? 😅